pkg/apis/apps/validation/validation.go

```diff
@@ -93,13 +93,6 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path) fi
 		}
 	case apps.RollingUpdateStatefulSetStrategyType:
 		if spec.UpdateStrategy.RollingUpdate != nil {
-			if spec.PodManagementPolicy == apps.OrderedReadyPodManagement &&
```

Comment:
this seems to undo changes added by your first commit, needs cleanup - https://github.com/kubernetes/kubernetes/pull/82162/commits/c69cb2e82fe2f6715fd28bb5bd5810aaad014ce2#diff-f4270171dae1b0ad1792c3f5c5ac451e823af053462f464dd38d32fec4846b08R96

Reply:
@tnozicka see this https://github.com/kubernetes/kubernetes/pull/82162#discussion_r523390143 .  Its now implemented for both orderedReady and parallel, so we dont need to error out in this scenario, so i have removed those checks